sdk/digitaltwins/Azure.DigitalTwins.Core/src/Models/BasicDigitalTwinComponentMetadata.cs

```diff
@@ -0,0 +1,25 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Collections.Generic;
+using System.Text.Json.Serialization;
+
+namespace Azure.DigitalTwins.Core
+{
+    /// <summary>
+    /// An optional, helper class for deserializing a digital twin.
+    /// The $metadata object on a <see cref="BasicDigitalTwinComponent"/>.
+    /// </summary>
+    /// <remarks>
+    /// For more samples, see <see href="https://github.com/Azure/azure-sdk-for-net/tree/master/sdk/digitaltwins/Azure.DigitalTwins.Core/samples">our repo samples</see>.
+    /// </remarks>
+    public class BasicDigitalTwinComponentMetadata
+    {
+        /// <summary>
+        /// This field will contain metadata about changes on properties on a component.
+        /// For your convenience, the value can be deserialized into <see cref="BasicDigitalTwinPropertyMetadata"/>.
+        /// </summary>
+        [JsonExtensionData]
+        public IDictionary<string, object> PropertyMetadata { get; set; } = new Dictionary<string, object>();
```

Comment:
We didn't come up with a name for this dictionary. Thoughts @johngallardo?

Reply:
I think PropertyMetadata is fine.

Reply:
But... see my comment in BasicDigitalTwinComponent --- I think we can actually get rid of this type.